File: countPrimes.java
class Solution {
  public int countPrimes(int n) {
    if (n <= 1) {
      return 0;
    }

    int[] primes = new int[n];
    int count = 0;

    for (int i = 2; i < n; i++) {
      if (primes[i] == 0) {
        primes[i] = 1;
        count++;

        for (int j = i * 2; j < n; j += i) {
          primes[j] = -1;
        }
      }
    }

    return count;
  }
}

// test cases

// n = 10
// n = 0
// n = 1

File: countPrimes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 1:
            return 0

        primes = [0 for _ in range(n)]
        count = 0

        for i in range(2, n):
            if primes[i] == 0:
                primes[i] = 1
                count += 1

                for j in range(i * 2, n, i):
                    primes[j] = -1

        return count